datasets/flwr_datasets/partitioner/continuous_partitioner.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Continuous partitioner class that works with Hugging Face Datasets."""
+
+
+# pylint: disable=R0913, R0917
+from typing import Optional
+
+import numpy as np
+
+from datasets import Dataset
+from flwr_datasets.partitioner import Partitioner
+
+
+class ContinuousPartitioner(
+    Partitioner
+):  # pylint: disable=too-many-instance-attributes
+    """Partitioner based on a continuous dataset property with adjustable strictness.
+
+    This partitioner enables non-IID partitioning by sorting the dataset based on a
+    continuous property and introducing Gaussian noise controlled by a strictness parameter.
```

Comment:
The _" and introducing Gaussian noise controlled by a strictness parameter."_ isn't very easy to understand for someone finding for the first time this partitioner. Maybe you can provide a simple example of what this would mean?